### ISSUE TITLE
 Wro4jTag threw an exception; see logs for details

### ISSUE THREAD

**Issue by rraman** [alf239/wro4j-tag#1]
Hello,

I/m trying to use tag lib for one of my project. But getting the below error:

```
        at uk.ac.ebi.gxa.web.wro4j.tag.Wro4jTag.jspTagException(Wro4jTag.java:103)
        at uk.ac.ebi.gxa.web.wro4j.tag.Wro4jTag.doStartTag(Wro4jTag.java:85)
```

On digging further, I found out the exception is caused by the statement:

```
    if (CURRENT.get() == null) {
        throw new WroRuntimeException("No context associated with CURRENT request cycle!");
    }
```

in `validateContext()` of `ro.isdc.wro.config.Context` class. The reason being, the context is being set in the thread local only in the `WroFilter`. But when including the tag lib, the filter won't me invoked at all.

How to resolve this issue?

Regards,
Rajaraman

**Comment by alf239** [comment]
Hi Rajaraman, thank you for the issue report — we're looking into it.

**Comment by alf239** [comment]
_Update: not related actually_ From a quick search, may be related to http://code.google.com/p/wro4j/issues/detail?id=61

**Comment by alf239** [comment]
@rraman, could you please add a full stacktrace printed by the following statement?

```
 log.error("Wro4jTag error: " + e.getMessage(), e);
```

It's line 102 of `Wro4jTag`:

```
    private JspTagException jspTagException(Exception e) {
        log.error("Wro4jTag error: " + e.getMessage(), e);
        return new JspTagException("Wro4jTag threw an exception; see logs for details");
    }
```

**Comment by alexo** [comment]
Hi, actually the referred link is not related to the problem described in this issue (the [issue61](http://code.google.com/p/wro4j/issues/detail?id=61) was related to periodic cache reload and was fixed long time ago). 

The reported problem has a valid point, since `wro4j` requires a context being stored in a thread local, which is not available when being used as taglib. Probably a possible solution could be setting the context explicitly.

Btw, I would love to add taglib to `wro4j-core` (as part of the library), and this project has a lot of great stuff I would like to reuse.

**Comment by rraman** [comment]
Yes you are right, `wro4j` needs a context set in the thread local. I made the tag work by doing

``` try{
Context.set(Context.webContext(request, (HttpServletResponse)pageContext.getResponse(), <filterConfig>), wroConfiguration)
//tag handling
} finally {
Context.unset()
}
```

Of course, I did some ugly patching in `wro4j` my making the `filterConfig` & `wroConfiguration` static. 

Also, why there is a need for separate `wro4j-tag properties`? We just utilize the `wro4j.properties` which is already configured.

Update: Just found out from response by Olga Melnichuk, `wro4j-tag` is a buildtime tool. But I would like to use it for the runtime, as ours is the saas product and we dont build the project quite often.
